Modules/TPC/src/TH2ReductorTPC.cxx

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file     TH2Reductor.cxx
+/// \author   Marcel Lesch
+/// \author   Cindy Mordasini
+/// \author   Based on the work from Piotr Konopka
+///
+
+#include "QualityControl/QcInfoLogger.h"
+#include "TPC/TH2ReductorTPC.h"
+#include <vector>
+#include <TAxis.h>
+#include <TCanvas.h>
+#include <TH2.h>
+#include <TList.h>
+
+namespace o2::quality_control_modules::tpc
+{
+void TH2ReductorTPC::update(TObject* obj, std::vector<SliceInfo>& reducedSource,
+                            std::vector<std::vector<float>>& axis)
+{
+  // Define the local variables in the default case: 1 single pad
+  // (no multipad canvas, nor slicer), and slicer axes size set to 1 (no slicing).
+  TList* padList = nullptr; // List of TPads if input TCanvas.
+  TH2* histo = nullptr;     // Pointer to the histogram to trend.
+  int numberPads = 1;       // Number of output objects.
+  int axisSize = 1;         // Default value for the main slicer axis.
+  int innerAxisSize = 1;    // Default value for the inner slicer axis.
+
+  // Get the number of pads, and their list in case of an input canvas.
+  const bool isCanvas = (obj->IsA() == TCanvas::Class());
+  if (isCanvas) {
+    auto canvas = static_cast<TCanvas*>(obj);
+    padList = static_cast<TList*>(canvas->GetListOfPrimitives());
+    padList->SetOwner(kTRUE);
+    numberPads = padList->GetEntries();
+  } else {                    // Slicer case: number of pads = axis.size*(innerAxis.size-1).
+    if (axis[0].size() > 1) { // Ensure the slicer config is valid.
+      axisSize = (int)axis.size();
+      innerAxisSize = (int)axis[0].size() - 1;
+    }
+  }
```

Comment:
Same comments as to TH1 apply here:
- Clarify the cases in the comments
- Add an "else" for the default case (no canvas, no slicing)